### DISCUSSION TITLE
How to get the self-intersection location when the LineString self-intersection

### DISCUSSION THREAD

**Discussion by wqshen** [shapely/shapely#1450] [Q&A]
The LineString class has is_valid attribute, but can i get the intersection point when it self-intersection ?

**Comment by sgillies** [comment]
`explain_validity()` (http://toblerity.org/shapely/manual.html#diagnostics) is the closest thing we can offer right now. It will require you to scrape text for the self-intersection point, which is less than ideal.

**Comment by jmon12** [comment]
[Here](https://gis.stackexchange.com/questions/423351/identifying-self-intersections-in-linestring-using-shapely?noredirect=1&lq=1) is the best workaround I found: perform a `unary_union` and the duplicated points are the intersection points, e.g. something like that:

```python
from shapely.ops import unary_union
from shapely.geometry import Point, MultiPoint
from collections import Counter

def find_self_intersection(line):
    if not line.is_simple:
        intersection = unary_union(line)
        seg_coordinates = []
        for seg in intersection.geoms:
            seg_coordinates.extend(list(seg.coords))
        intersection = [Point(p) for p, c in Counter(seg_coordinates).items() if c > 1]
        intersection = MultiPoint(intersection)
        return intersection
```
